api/tests/opentrons/protocol_api/core/legacy/test_thermocycler_core.py

```diff
@@ -0,0 +1,477 @@
+    subject: LegacyThermocyclerCore,
+) -> None:
+    """It should report the current step index."""
+    decoy.when(mock_sync_module_hardware.current_step_index).then_return(42)
+    result = subject.get_current_step_index()
+    assert result == 42
+
+
+def test_open_lid(
+    decoy: Decoy,
+    mock_sync_module_hardware: SyncThermocyclerHardware,
+    mock_sync_hardware_api: SyncHardwareAPI,
+    mock_protocol_core: ProtocolContextImplementation,
+    mock_instrument_core: InstrumentContextImplementation,
+    mock_labware: Labware,
+    mock_well: Well,
+    subject: LegacyThermocyclerCore,
+) -> None:
+    """It should open the lid with the hardware."""
+    decoy.when(mock_protocol_core.get_loaded_instruments()).then_return(
+        {Mount.RIGHT: mock_instrument_core}
+    )
+    decoy.when(mock_instrument_core.get_mount()).then_return(Mount.RIGHT)
+    decoy.when(mock_sync_hardware_api.current_position(Mount.RIGHT)).then_return(
+        {Axis.A: 4}
+    )
+    decoy.when(subject._get_fixed_trash()).then_return(mock_labware)
+    decoy.when(mock_labware.wells()).then_return([mock_well])
+    decoy.when(mock_well.top()).then_return(
+        Location(point=Point(x=1, y=2, z=3), labware=mock_well)
+    )
+
+    decoy.when(mock_sync_module_hardware.open()).then_return("open")
+
+    result = subject.open_lid()
+
+    decoy.verify(
+        mock_sync_hardware_api.retract(Mount.RIGHT),
+        mock_instrument_core.move_to(
+            Location(Point(x=1, y=2, z=4), None),
+            force_direct=True,
+            minimum_z_height=None,
+            speed=None,
+        ),
+    )
+    assert result == "open"
+
+
+def test_close_lid(
+    decoy: Decoy,
+    mock_sync_module_hardware: SyncThermocyclerHardware,
+    mock_sync_hardware_api: SyncHardwareAPI,
+    mock_protocol_core: ProtocolContextImplementation,
+    mock_instrument_core: InstrumentContextImplementation,
+    mock_labware: Labware,
+    mock_well: Well,
+    subject: LegacyThermocyclerCore,
+) -> None:
+    """It should close the lid with the hardware."""
+    decoy.when(mock_protocol_core.get_loaded_instruments()).then_return(
+        {Mount.LEFT: mock_instrument_core}
+    )
+    decoy.when(mock_instrument_core.get_mount()).then_return(Mount.LEFT)
+    decoy.when(mock_sync_hardware_api.current_position(Mount.LEFT)).then_return(
+        {Axis.Z: 4}
+    )
+    decoy.when(subject._get_fixed_trash()).then_return(mock_labware)
+    decoy.when(mock_labware.wells()).then_return([mock_well])
+    decoy.when(mock_well.top()).then_return(
+        Location(point=Point(x=1, y=2, z=3), labware=mock_well)
+    )
+
+    decoy.when(mock_sync_module_hardware.close()).then_return("close")
+
+    result = subject.close_lid()
+
+    decoy.verify(
+        mock_sync_hardware_api.retract(Mount.LEFT),
+        mock_instrument_core.move_to(
+            Location(Point(x=1, y=2, z=4), None),
+            force_direct=True,
+            minimum_z_height=None,
+            speed=None,
+        ),
```

Comment:
We should also verify that the geometry state is updated. See Decoy docs about [property setter verification](https://mike.cousins.io/decoy/advanced/properties/#verifying-a-setter)

```suggestion
        ),
        decoy.prop(mock_geometry.lid_status).set("closed"),
```